tests/baselines/reference/moduleExportAssignment7.errors.txt

```diff
@@ -0,0 +1,69 @@
+tests/cases/conformance/salsa/index.ts(2,24): error TS2694: Namespace '"tests/cases/conformance/salsa/mod".export=' has no exported member 'Thing'.
+tests/cases/conformance/salsa/index.ts(3,24): error TS2694: Namespace '"tests/cases/conformance/salsa/mod".export=' has no exported member 'AnotherThing'.
+tests/cases/conformance/salsa/index.ts(4,24): error TS2694: Namespace '"tests/cases/conformance/salsa/mod".export=' has no exported member 'foo'.
+tests/cases/conformance/salsa/index.ts(5,24): error TS2694: Namespace '"tests/cases/conformance/salsa/mod".export=' has no exported member 'qux'.
+tests/cases/conformance/salsa/index.ts(6,24): error TS2694: Namespace '"tests/cases/conformance/salsa/mod".export=' has no exported member 'baz'.
+tests/cases/conformance/salsa/index.ts(18,31): error TS2694: Namespace '"tests/cases/conformance/salsa/mod".export=' has no exported member 'buz'.
+
+
+==== tests/cases/conformance/salsa/mod.js (0 errors) ====
+    class Thing  { x = 1 }
+    class AnotherThing { y = 2  }
+    function foo() { return 3 }
+    function bar() { return 4 }
+    /** @typedef {() => number} buz */
+    module.exports = {
+        Thing,
+        AnotherThing,
+        foo,
+        qux: bar,
+        baz() { return 5 }
+    }
+==== tests/cases/conformance/salsa/main.js (0 errors) ====
+    /**
+     * @param {import("./mod").Thing} a

```

Comment:
What happens when you do `typeof` in the JS file? Is it possible to get the constructor type?

Reply:
Yes. I added a test for that.